### PR TITLE
[frio] Like buttons improvement

### LINKED ISSUE #12859: Decouple activity buttons rendering from conversation rendering
I try some improvement for the frontend.

And i need a little hint, how to build new endpoints to query for example for the likes/dislikes/shares of a posting or a comment.

Something like https://<instance>/api/bla/foo/get_responses?item=123456789&verb=like (the id of posting or comment) 
which returns a snippet of html-code, i can add/replace per jquery on the right place.

### PR DESCRIPTION
**Pull request by xundeenergie**
This is followup from https://github.com/friendica/friendica/pull/12874

Had to do some cleanup in my commits, and the old PR got lost.

The last action was reducing indentation.

### CODE REVIEW

File: README-jakob.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Anleitung für die Erweiterungen
```
Comment (MrPetovan) [inline README-jakob.md]: Please don’t add these explanation in a file you add to the repository. These belong to the PR itself, not the code.

File: view/theme/frio/js/theme.js [hunk 2]
```diff
@@ -764,12 +764,108 @@ function htmlToText(htmlString) {
...
+		case 'like':
+			thumbsClass = 'fa-thumbs-up';
+			break;
+		case 'dislike':
+			thumbsClass = 'fa-thumbs-down';
+			break;
+		case 'announce':
+			thumbsClass = 'fa-retweet';
+	}
+	// remindert to remove the like-rotator from templates
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Please remove the rotator from the templates right away.

File: view/theme/frio/js/theme.js [hunk 3]
```diff
@@ -764,12 +764,108 @@ function htmlToText(htmlString) {
...
+		$('button[id^=announce-' + ident.toString() + '] i:first-child').removeClass('fa-retweet');
+		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
+		if ($('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').length == 0) {
+			// append rotator to the shareMenu-button for small media
+			$('<img id="waitfor-' + verb + '-' + ident.toString() + '">')
+				.attr('src', 'images/rotator.gif')
+				.addClass('fa')
+				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' )).show();
+		} else {
+			$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').show()
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Please use the [`.attr(attributes)`](https://api.jquery.com/attr/#attr-attributes) form to set the `id` attribute along the `src` one and avoid writing literal HTML as much as possible
```suggestion
			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: view/theme/frio/js/theme.js (hunk 2 above)
Comment (xundeenergie) [reply view/theme/frio/js/theme.js]: Done

File: README-jakob.md (hunk 1 above)
Comment (xundeenergie) [reply README-jakob.md]: Ok I remove the README-jakob.md file.

**Comment by xundeenergie** [comment]
I did the changes for the errormessages in the 2 files, you named me (src/App/Page.php and view/theme/frio/templates/js_strings.tpl)
But i don't know, how to translate them. The strings.php, messages.po look all a bit redundant. Please give me a hint, how to translate such text.

File: view/theme/frio/js/theme.js [hunk 4]
```diff
@@ -784,8 +784,8 @@ function doActivityItemAction(ident, verb, un) {
 		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
 		if ($('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').length == 0) {
 			// append rotator to the shareMenu-button for small media
-			$('<img id="waitfor-' + verb + '-' + ident.toString() + '">')
-				.attr('src', 'images/rotator.gif')
+      $('<img>')
+        .attr('id', 'waitfor-' + verb + '-' + ident.toString())
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
			$('<img>')
				.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```

File: view/theme/frio/js/theme.js [hunk 5]
```diff
@@ -784,8 +784,8 @@ function doActivityItemAction(ident, verb, un) {
...
 				.addClass('fa')
 				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' )).show();
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Is the `show()` call even necessary?
```suggestion
				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' ))
				.show();
```

File: view/theme/frio/js/theme.js [hunk 6]
```diff
@@ -795,7 +795,8 @@ function doActivityItemAction(ident, verb, un) {
 	$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').removeClass(thumbsClass);
 	// if verb is announce, then one rotator is added above to the shareMedia-dropdown button
 	if ($('button:not(button.dropdown-toggle) img#waitfor-' + verb + '-' + ident.toString()).length == 0) {
-		$('<img id="waitfor-' + verb + '-' + ident.toString() + '">')
+		$('<img>')
+      .attr('id', 'waitfor-' + verb + '-' + ident.toString())
 			.attr('src', 'images/rotator.gif')
 			.addClass('fa')
 			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child')).show();
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Is the `show()` call necessary?
```suggestion
			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child'))
			.show();
```

File: view/theme/frio/js/theme.js [hunk 7]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+		} else {
+			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
+		}
+	}
+	$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').removeClass(thumbsClass);
+	// if verb is announce, then one rotator is added above to the shareMedia-dropdown button
+	if ($('button:not(button.dropdown-toggle) img#waitfor-' + verb + '-' + ident.toString()).length == 0) {
+		$('<img>')
+      .attr('id', 'waitfor-' + verb + '-' + ident.toString())
+			.attr('src', 'images/rotator.gif')
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
			.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```

File: view/theme/frio/js/theme.js [hunk 8]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+	.success(function(data){
+		//$('#like-rotator-' + ident.toString()).hide();
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+		if (data.status == 'ok') {
+			// response from server was ok
+			if (data.verb == 'un' + verb) {
+				// like/dislike buttons
+				$('button[id^=' + verb + '-' + ident.toString() + ']' )
+					.removeClass('active')
+					.attr('onclick', 'doActivityItemAction(' + ident +', "' + verb + '",false )').change();
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: What's the `change()` call doing in this case?
```suggestion
					.attr('onclick', 'doActivityItemAction(' + ident +', "' + verb + '",false )')
					.change();
```

File: view/theme/frio/js/theme.js [hunk 9]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+				// link in share-menu
+				$('a[id^=' + verb + '-' + ident.toString() + ']' )
+					.removeClass('active')
+					.attr('href', 'javascript:doActivityItemAction(' + ident +', "' + verb + '",false )').change();
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
					.attr('href', 'javascript:doActivityItemAction(' + ident +', "' + verb + '",false )')
					.change();
```

File: view/theme/frio/js/theme.js [hunk 10]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+				$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child' ).addClass('fa-retweet').removeClass('fa-ban');
+			} else {
+				// like/dislike buttons
+				$('button[id^=' + verb + '-' + ident.toString() + ']' )
+					.addClass('active')
+					.attr('onclick', 'doActivityItemAction(' + ident + ', "' + verb + '", true )').change();
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
					.attr('onclick', 'doActivityItemAction(' + ident + ', "' + verb + '", true )')
					.change();
```

File: view/theme/frio/js/theme.js [hunk 11]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+				// link in share-menu
+				$('a[id^=' + verb + '-' + ident.toString() + ']' )
+					.addClass('active')
+					.attr('href', 'javascript:doActivityItemAction(' + ident + ', "' + verb + '", true )').change();
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
					.attr('href', 'javascript:doActivityItemAction(' + ident + ', "' + verb + '", true )')
					.change();
```

File: view/theme/frio/js/theme.js [hunk 12]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+			/* server-response was not ok. Database-problems or some changes in
+			 * data?
+			 * reset all buttons
+			 */
+			$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+			$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').addClass('fa-share');
+			$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+			$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+			$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+      $.jGrowl(aActErr[verb] + '<br>(' + aErrType['srvErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Duplicated line and indentation.
```suggestion
			$.jGrowl(aActErr[verb] + '<br>(' + aErrType['srvErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```

File: view/theme/frio/js/theme.js [hunk 13]
```diff
@@ -764,12 +764,106 @@ function htmlToText(htmlString) {
...
+		}
+	})
+	.error(function(data){
+			/* Server could not be reaches successfully */
+			$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+			$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').addClass('fa-share');
+			$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+			$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
+      $.jGrowl(aActErr[verb] + '<br>(' + aErrType['netErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
		// Server could not be reached successfully
		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
		$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').addClass('fa-share');
		$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
		$('a[id^=' + verb + '-' + ident.toString() + '] i:first-child').addClass(thumbsClass);
		$.jGrowl(aActErr[verb] + '<br>(' + aErrType['netErr'] + ')', {sticky: false, theme: 'info', life: 5000});
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
For translation string changes, once you're done with your code changes, you need to run the following script from the Friendica root folder: `bin/run_xgettext.sh`. This will update the relevant translation files and you just have to commit the produced changes.

File: view/theme/frio/js/theme.js (hunk 6 above)
Comment (xundeenergie) [reply view/theme/frio/js/theme.js]: not neccessary. removed it.

**Comment by xundeenergie** [comment]
are there any open requests? 
I can not see them... 
Will this PR go to 2023.03-rc too?

**Comment by tobiasd** [comment]
Could you please remove the README-jakob.md file

**Comment by xundeenergie** [comment]
> Could you please remove the README-jakob.md file

done!

File: view/theme/frio/js/theme.js [hunk 14]
```diff
@@ -764,12 +764,103 @@ function htmlToText(htmlString) {
...
+		$('button[id^=announce-' + ident.toString() + '] i:first-child').removeClass('fa-retweet');
+		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
+		if ($('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').length == 0) {
+			// append rotator to the shareMenu-button for small media
+			$('<img>')
+				.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
+				.addClass('fa')
+				.appendTo($('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child' ));
+		} else {
+			$('<img>').attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: This creates an element but never attaches it to the page DOM, is it intentional?

File: view/theme/frio/js/theme.js [hunk 15]
```diff
@@ -764,12 +764,103 @@ function htmlToText(htmlString) {
...
+	$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').removeClass(thumbsClass);
+	// if verb is announce, then one rotator is added above to the shareMedia-dropdown button
+	if ($('button:not(button.dropdown-toggle) img#waitfor-' + verb + '-' + ident.toString()).length == 0) {
+		$('<img>')
+			.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
+			.addClass('fa')
+			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child'));
+	} else {
+		// show existing rotator for activity
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']')
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: This selects an image but never does anything with it, is it intentional?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: view/theme/frio/js/theme.js [hunk 16]
```diff
@@ -764,12 +764,97 @@ function htmlToText(htmlString) {
...
 	}
-	$("#" + verb + "-" + ident).toggleClass("active");
-
-	doActivityItem(ident, verb, un);
+	if (verb.indexOf('announce') === 0 ) {
+		// Share-Button(s)
+		// remove share-symbol, to replace it by rotator
+		$('button[id^=shareMenuOptions-' + ident.toString() + '] i:first-child').removeClass('fa-share');
+		$('button[id^=announce-' + ident.toString() + '] i:first-child').removeClass('fa-retweet');
+		// if no wait-rotator for activity(verb) is added, add it. or just show it, if exists
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]:
```suggestion
		// if no wait-rotator exists for activity(verb), add it.
```

File: view/theme/frio/js/theme.js [hunk 17]
```diff
@@ -764,12 +764,97 @@ function htmlToText(htmlString) {
...
+	}
+	$('button[id^=' + verb + '-' + ident.toString() + '] i:first-child').removeClass(thumbsClass);
+	// if verb is announce, then one rotator is added above to the shareMedia-dropdown button
+	if ($('button:not(button.dropdown-toggle) img#waitfor-' + verb + '-' + ident.toString()).length == 0) {
+		$('<img>')
+			.attr({id: 'waitfor-' + verb + '-' + ident.toString(), src: 'images/rotator.gif'})
+			.addClass('fa')
+			.appendTo($('button[id^=' + verb + '-' + ident.toString() + '] i:first-child'));
+	}
+	// do request for activity
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Redundant with code
```suggestion

```

File: view/theme/frio/js/theme.js [hunk 18]
```diff
@@ -764,12 +764,97 @@ function htmlToText(htmlString) {
...
+	$.post('item/' + ident.toString() + '/activity/' + _verb)
+	.success(function(data){
+		$('img[id^=waitfor-' + verb + '-' + ident.toString() + ']').remove();
+		if (data.status == 'ok') {
+			// response from server was ok
```
Comment (MrPetovan) [inline view/theme/frio/js/theme.js]: Comment redundant with code.
```suggestion
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]
Last touches.

File: view/theme/frio/js/theme.js (hunk 16 above)
Comment (xundeenergie) [reply view/theme/frio/js/theme.js]: changed line to
"avoid multiple rotators on like/share-button if klicked multiple times"

**Review by MrPetovan — APPROVED** [review]

**Comment by MrPetovan** [comment]
Thank you for your work!

**Comment by xundeenergie** [comment]
> Thank you for your work!

Thank you for your attendation and understanding a newcomer :)

**Comment by MrPetovan** [comment]
Closes #12859

**Pull request merged**